Source/main.cpp

```diff
@@ -34,6 +34,11 @@ int main(int argc, char* argv[])
 
     amrex::Initialize(argc,argv);
 
+    // we need a larger stack limit that usual bc of the parser.
+#if defined(AMREX_USE_GPU) && defined(AMREX_USE_CUDA)
+    AMREX_CUDA_SAFE_CALL(cudaDeviceSetLimit(cudaLimitStackSize, 20*1024));
```

Comment:
I am trying to spot the default size on a P100/V100/A100... Do you know?
https://docs.nvidia.com/cuda/cuda-c-programming-guide/index.html#features-and-technical-specifications

Does increasing this generally lead to a performance drop, e.g. due to limited parallelism?

Reply:
I am worried about it too.

Reply:
The runtime on the uniform plasma benchmark we have been using is exactly the same as if I hack out the parser stuff altogether, so that I can leave the limit alone.

Reply:
The runtime on the uniform plasma benchmark we have been using is exactly the same as if I hack out the parser stuff altogether, so that I can leave the limit alone.

Reply:
Do you know where the rest of the memory goes to? Is it shared with L1 or shared memory or so? Have not turned that knob before.

Reply:
Update: ah, with the Debug update it does not really matter now. Now I am just curious (:

Reply:
paraphrasing @maxpkatz for the sake of googling the solution:

This is memory reserved for thread-local stack memory. The amount of reserved stack memory is stack memory per thread * number of (potentially) possible resident threads on the device.

The `cudaLimitStackSize` is usually about half of the pre-reserved 300MByte (on a V100) and stored in device DRAM.
If one doubles the stack size, one will double the amount of memory reserved (so one starts with 450 MB gone).

:+1: all clear now.